### PR TITLE
Die on mksquashfs failure.

### PR DESCRIPTION
**Pull request by lalten**
This PR adds code to check the return code of the mksquashfs process. appimagekit will now die if mksquashfs failed.

### CODE REVIEW

File: src/appimagetool.c [hunk 1]
```diff
@@ -131,13 +131,26 @@ int sfs_ls(char* image) {
 * execlp(), execvp(), and execvpe() search on the $PATH */
 int sfs_mksquashfs(char *source, char *destination, int offset) {
     pid_t pid = fork();
-    
     if (pid == -1) {
-        // error, failed to fork()
+        perror("fork()");
```
Comment (TheAssassin) [inline src/appimagetool.c:135]: Please add some context information. We use other subprocesses, too, within the scope of AppImageKit. Ideally, the message informs the user that this was trying to open the `mksquashfs` process.

**Review by TheAssassin — CHANGES_REQUESTED** [review]
Thanks, looks good overall. See my comment, applies to all `perror` calls.

**Comment by lalten** [comment]
ping on this one as well @TheAssassin :)

**Comment by TheAssassin** [comment]
Will have a look. Haven't had time for AppImage related projects in a while, and when things broke I had to invest like a full week or maybe even two...

File: src/appimagetool.c [hunk 2]
```diff
@@ -131,13 +131,26 @@ int sfs_ls(char* image) {
...
         int status;
-        waitpid(pid, &status, 0);
+        if(waitpid(pid, &status, 0) == -1) {
+            perror("sfs_mksquashfs waitpid() failed");
+            return(-1);
+        }
+        
+        int retcode = WEXITSTATUS(status);
+        if (retcode) {
+            fprintf(stderr, "mksquashfs (pid %ld) exited with code %d\n", (long)pid, retcode);
```
Comment (TheAssassin) [inline src/appimagetool.c:149]: `pid_t` should be an `int` in glibc, so I'm not sure this cast is necessary, `%d` should work. (After all, `long int` and `int` are most likely the same size (4 bytes) on every viable platform anyway...)

**Comment by lalten** [comment]
Friendly ping on this one, @TheAssassin :)

**Review by TheAssassin — APPROVED** [review]

**Comment by TheAssassin** [comment]
Thanks!

**Pull request merged**